pennylane/transforms/defer_measurements.py

```diff
@@ -48,11 +48,46 @@ def defer_measurements(tape: QuantumTape) -> (Sequence[QuantumTape], Callable):
 
     .. note::
 
-        Devices that inherit from :class:`~pennylane.QubitDevice` **must** be initialized
-        with an additional wire for each mid-circuit measurement after which the measured
-        wire is reused or reset for ``defer_measurements`` to transform the quantum tape
-        correctly. Hence, devices and quantum tapes must also be initialized without custom
-        wire labels for correct behaviour.
+        Devices that inherit from :class:`~pennylane.QubitDevice` such as
+        ``default.qubit.legacy`` **must** be initialized with an additional wire for
+        each mid-circuit measurement after which the measured wire is reused or reset
+        for ``defer_measurements`` to transform the quantum tape correctly. Devices
+        that inherit from :class:`~pennylane.devices.Device` such as ``default.qubit``
+        must be initialized with the same number of wires, or with ``wires=None``.
+
+    .. note::
+
+        If using custom wire labels, do not use ``"mv{i}"`` as labels, where ``{i}`` is an
+        integer. These labels are reserved for wires that store mid-circuit measurement results
+        if transforming a tape or executing on a device that has ``wires=None``.
```

Comment:
Could we rephrase this as:
- If you are specifying wires when loading your device, you'll need to account for the additional wires due to qubit reuse and reset
- If you don't, we'll use `"mv{i}"` for you